include/boost/uuid/string_generator.hpp

```diff
@@ -122,23 +133,27 @@ struct string_generator {
         static char const*const digits_end = digits_begin + 22;
```

Comment:
This isn't your code but can't be take the sizeof(digits_begin) and use that instead of hardcoding 22 and further on down hardcoding '> 21' ? Hardcoding numbers is generally poor c++ if there is a better way to get the hardcoded value from the compiler.